commit c0463502ff89db15c9bcfee33878678b7fd14cef
Author: Paul Bakker <p.j.bakker@polarssl.org>
Date:   Thu Feb 14 11:19:38 2013 +0100

Fixed memory leak in ssl_free() and ssl_reset() for active session

--- library/ssl_tls.c
+++ library/ssl_tls.c
@@ -3025,6 +3025,13 @@ int ssl_session_reset( ssl_context *ssl )
         ssl->transform = NULL;
     }
 
+    if( ssl->session )
+    {
+        ssl_session_free( ssl->session );
+        free( ssl->session );
+        ssl->session = NULL;
+    }
+
     if( ( ret = ssl_handshake_init( ssl ) ) != 0 )
         return( ret );
 
@@ -3891,6 +3898,12 @@ void ssl_free( ssl_context *ssl )
         free( ssl->session_negotiate );
     }
 
+    if( ssl->session )
+    {
+        ssl_session_free( ssl->session );
+        free( ssl->session );
+    }
+
     if ( ssl->hostname != NULL)
     {
         memset( ssl->hostname, 0, ssl->hostname_len );
